README.md

```diff
@@ -17,6 +17,7 @@
   - [Contracts](#contracts)
   - [Object-Oriented Programming](#object-oriented-programming)
   - [Functional Programming](#functional-programming)
+  - [Pattern Matching](#pattern-matching)
```

Comment:
The old one needs to be removed.

Reply:
The pattern matching category in the index reflects the pattern matching category in real, exactly under the functional programming category.

I wasnt sure, if I should merge them, considering that many Nim users seem to see it on it's own.

Why would you remove that index, but not the real category way down?